virtualizarr/manifests/array_api.py

```diff
@@ -34,12 +34,17 @@ def decorator(func):
 
 
 @implements(np.result_type)
-def result_type(*arrays_and_dtypes) -> np.dtype:
+def result_type(*arrays_and_dtypes: Union["ManifestArray", np.dtype]) -> np.dtype:
```

Comment:
This was an opportunistic refactor and I should move this change (and the new tests) to a separate PR.

Reply:
#678